### PR TITLE
Refactor: Migrate `StakingProductsCardGrid` to Chakra

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
## Description
Migrate the `StakingProductsCardGrid` component to Chakra UI

>Side Note: This PR also fixes broken styling present in the current production.

## Related Issue
#8632

### CODE REVIEW

**Review by pettinarip — CHANGES_REQUESTED** [review]
This looks great @TylerAPfledderer! left a comment.

File: src/components/Staking/StakingProductsCardGrid.tsx [hunk 1]
```diff
@@ -303,48 +239,89 @@ const StakingProductCard: React.FC<ICardProps> = ({
...
+      <HStack
+        {...PADDED_DIV_STYLE}
+        spacing={6}
+        background={color}
+        bgGradient="linear(0deg, rgba(0, 0, 0, 30%), rgba(0, 0, 0, 0))"
+        borderRadius="base"
+        maxH={24}
+      >
+        {!!Svg && <Icon as={Svg} fontSize="2rem" />}
+        <Heading fontSize="2xl" color="white" fontWeight="normal">
```
[the remark this answers is not in the document]
Comment (TylerAPfledderer) [reply src/components/Staking/StakingProductsCardGrid.tsx:260]: Got it. Agreed, I can see a possible quick alteration with the grid to overcome this text wrapping, but it's definitely better with it's own PR and design team input.

[the remark this answers is not in the document]
Comment (TylerAPfledderer) [reply src/components/Staking/StakingProductsCardGrid.tsx]: Change of font weight was not intended to fix the header height. I removed the `fontWeight` prop.

**Review by pettinarip — APPROVED** [review]
LGTM! thanks @TylerAPfledderer

**Pull request merged**